### ISSUE TITLE
[Suggestion] Let items with components be used as currency, separate from the default item

### ISSUE THREAD

**Issue by callyral** [IAmSneak/diamond-economy#30]
This would allow for items with specific names (and components in general) to be used as currency (something like `minecraft:diamond[minecraft:custom_name='{"text":"Super Diamond"}']`).

This feature would also be useful for creating currency items only obtainable by admins or by using command blocks.

Currently, the database just stores the item ID (or seems to, I looked at it using [sqlitebrowser](https://github.com/sqlitebrowser/sqlitebrowser)), so even if you specify an item with components, you'll just be able to use the default item as a currency anyways. The command I used to test this is `/diamonds currency add minecraft:gold_ingot[minecraft:custom_name='{"text":"coin","italic":false}'] 1 true 1 true true`.

It would also be nice to have a way to specific an item with components as a main currency in the config file. IIRC components are a way to represent JSON data, so using them in the config with a JSON object of some kind feels like a nice solution.

[Relevant minecraft.wiki page](https://minecraft.wiki/w/Data_component_format)

**Comment by IAmSneak** [comment]
i agree this would be nice, up to future me whether i implement it or not
